BPA/Ch.10/js_ms.stan

```diff
@@ -16,133 +16,129 @@ data {
-            * po[k, i, t - 1, y[i, t]];
-        gam[t, k] <- sum(acc);
+          acc[j] = gam[t - 1, j] * ps[j, i, t - 1, k]
+                  * po[k, i, t - 1, y[i, t]];
+        gam[t, k] = sum(acc);
       }
     }
-    increment_log_prob(log(sum(gam[n_occasions])));
+    target += log(sum(gam[n_occasions]));
   }
 }
 
 generated quantities {
-  real<lower=0,upper=1> psi;              // Inclusion probability
-  real<lower=0,upper=1> b[n_occasions-1]; // Entry probability
-  int<lower=0> Nsuper;                    // Superpopulation size
-  int<lower=0> N[n_occasions-1];          // Actual population size
-  int<lower=0> B[n_occasions-1];          // Number of entries
-  int<lower=1,upper=3> z[M, n_occasions]; // Latent state
+  real<lower=0,upper=1> psi;                // Inclusion probability
+  vector<lower=0,upper=1>[n_occ_minus_1] b; // Entry probability
+  int<lower=0> Nsuper;                      // Superpopulation size
+  int<lower=0> N[n_occ_minus_1];            // Actual population size
+  int<lower=0> B[n_occ_minus_1];            // Number of entries
+  int<lower=1,upper=3> z[M, n_occasions];   // Latent state
 
   // Generate z[]
   for (i in 1:M) {
-    z[i, 1] <- 1;
+    z[i, 1] = 1;
     for (t in 2:n_occasions)
-      z[i, t] <- categorical_rng(ps[z[i, t - 1], i, t - 1]);
+      z[i, t] = categorical_rng(ps[z[i, t - 1], i, t - 1]);
   }
 
   // Calculate derived population parameters
   {
-    real qgamma[n_occasions-1];
-    real cprob[n_occasions-1];
-    int al[M, n_occasions-1];
-    int d[M, n_occasions-1];
+    vector[n_occ_minus_1] qgamma;
+    vector[n_occ_minus_1] cprob;
+    int al[M, n_occ_minus_1];
+    int d[M, n_occ_minus_1];
     int alive[M];
     int w[M];
 
-    for (t in 1:(n_occasions - 1)) {
-      qgamma[t] <- 1.0 - gamma[t];
-    }
-    cprob[1] <- gamma[1];
-    for (t in 2:(n_occasions - 1)) {
-      cprob[t] <- gamma[t] * prod(qgamma[1:(t - 1)]);
-    } //t
-    psi <- sum(cprob);
-    for (t in 1:(n_occasions - 1)) {
-      b[t] <- cprob[t] / psi;
-    } //t
+    qgamma = 1.0 - gamma;
+    cprob[1] = gamma[1];
+    for (t in 2:n_occ_minus_1)
+      cprob[t] = gamma[t] * prod(qgamma[1:(t - 1)]);
+    psi = sum(cprob);
+    b = cprob / psi;
 
     for (i in 1:M) {
       for (t in 2:n_occasions)
-        al[i, t - 1] <- (z[i, t] == 2);
-      for (t in 1:(n_occasions - 1)) {
-        if (z[i, t] - al[i, t] == 0)
-          d[i, t] <- 1;
-        else
-          d[i, t] <- 0;
-      } //t
-      alive[i] <- sum(al[i]);
-    } //i
-
-    for (t in 1:(n_occasions - 1)) {
-      N[t] <- sum(al[1:M, t]);
-      B[t] <- sum(d[1:M, t]);
-    } //t
+        al[i, t - 1] = (z[i, t] == 2);
+      for (t in 1:n_occ_minus_1)
+        d[i, t] = (z[i, t] - al[i, t] == 0);
+      alive[i] = sum(al[i]);
+    }
+
+    for (t in 1:n_occ_minus_1) {
+      N[t] = sum(al[, t]);
+      B[t] = sum(d[, t]);
+    }
     for (i in 1:M)
-      w[i] <- (alive[i] != 0);
-    Nsuper <- sum(w);
+      w[i] = (alive[i] != 0);
```

Comment:
This can just be `!alive[i]` --- you usually don't want to be comparing booleans to 1 and 0.  Sorry I didn't catch a lot of this the first review.  There's always something to make better with code.